File: 1. Two Sum.cpp
class Solution {
public:
vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> um;
    int n=nums.size();
        vector<int> v;
        for(int i=0; i<n; i++){
            if(um.find(target-nums[i])!= um.end()){
                v.push_back(i);
                v.push_back(um[target-nums[i]]);
                break;
            }
            um[nums[i]]= i;
        }
        return v;
    }
};

File: Leetcode/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idx={}
        idx[nums[0]]=0
        for i in range(1,len(nums)):
            if target-nums[i] in idx:
                return [idx[target-nums[i]],i]
            else:
                idx[nums[i]]=i
        return []
        
